drivers/tee/optee/core.c

```diff
@@ -393,7 +459,11 @@ optee_config_shm_memremap(optee_invoke_fn *invoke_fn, void **memremaped_shm)
 	dmabuf_info.paddr = paddr + OPTEE_SHM_NUM_PRIV_PAGES * PAGE_SIZE;
 	dmabuf_info.size = size - OPTEE_SHM_NUM_PRIV_PAGES * PAGE_SIZE;
 
-	pool = tee_shm_pool_alloc_res_mem(&priv_info, &dmabuf_info);
+	/* If OP-TEE can work with unregistered SHM, we will use own pool */
+	if (sec_caps & OPTEE_SMC_SEC_CAP_DYNAMIC_SHM)
+		pool = optee_shm_get_pool(&dmabuf_info);
```

Comment:
Aha, sorry I didn't realize that the already present pool doesn't allocate command buffers correctly. Please keep the "own pool".

In the meantime I'll try to figure out how this can work together with the already present pool instead of replacing it with partially duplicated code.